app/src/main/java/com/keylesspalace/tusky/util/SpanUtils.kt

```diff
@@ -100,16 +116,15 @@ private fun findEndOfPattern(string: String, result: FindCharsResult, pattern: P
 
 private fun getSpan(matchType: FoundMatchType, string: String, colour: Int, start: Int, end: Int): CharacterStyle {
     return when(matchType) {
-        FoundMatchType.HTTP_URL -> CustomURLSpan(string.substring(start, end))
-        FoundMatchType.HTTPS_URL -> CustomURLSpan(string.substring(start, end))
+        FoundMatchType.URL -> CustomURLSpan(string.substring(start, end))
         else -> ForegroundColorSpan(colour)
     }
 }
 
 /** Takes text containing mentions and hashtags and urls and makes them the given colour. */
-fun highlightSpans(text: Spannable, colour: Int) {
+fun highlightSpans(text: Spannable, colour: Int, spanClassesToStrip: List<Class<out Any>> = spanClasses) {
```

Comment:
I'm struggling to understand some f these tbh
you still clear URL spans?.. and check for existing ones later?..
I would be glad if you explained it and maybe we could figure out how to document it together.

Reply:
Sure. In `ComposeActivity`, we still strip all spans, because we're modifying the text.
In `LinkHelper`, we don't strip any spans; we just add new ones if necessary and set the appropriate link listener on existing ones.

Reply:
Thanks, now I understand!
I'll have a look again later, maybe this weekend.
Btw, CI fails for some reason?

Reply:
Hm, I forgot that some of the highlighting tests are in `ComposeActivityTest` and missed those. They look like legitimate failures; I'll investigate.